daffodil-lib/src/main/scala/org/apache/daffodil/api/WarnID.scala

```diff
@@ -21,14 +21,33 @@ import org.apache.daffodil.exceptions.ThrowsSDE
 import org.apache.daffodil.exceptions.Assert
 import org.apache.daffodil.schema.annotation.props.{ Enum => PropsEnum }
 
-  sealed trait WarnID extends WarnID.Value
+sealed trait WarnID extends WarnID.Value
 
-  object WarnID extends PropsEnum[WarnID] {
-    case object All extends WarnID; forceConstruction(All)
-    case object MultipleChoiceBranches extends WarnID; forceConstruction(MultipleChoiceBranches)
-    case object EscapeSchemeRefUndefined extends WarnID; forceConstruction(EscapeSchemeRefUndefined)
+object WarnID extends PropsEnum[WarnID] {
+  case object All extends WarnID; forceConstruction(All)
+  case object MultipleChoiceBranches extends WarnID; forceConstruction(MultipleChoiceBranches)
+  case object EscapeSchemeRefUndefined extends WarnID; forceConstruction(EscapeSchemeRefUndefined)
+  case object NoEmptyDefault extends WarnID; forceConstruction(NoEmptyDefault)
+  case object AppinfoNoSource extends WarnID; forceConstruction(AppinfoNoSource)
+  case object AppinfoDFDLSourceWrong extends WarnID; forceConstruction(AppinfoDFDLSourceWrong)
+  case object QueryStylePathExpression extends WarnID; forceConstruction(QueryStylePathExpression)
+  case object PatternEncodingSlashW extends WarnID; forceConstruction(PatternEncodingSlashW)
+  case object NamespaceDifferencesOnly extends WarnID; forceConstruction(NamespaceDifferencesOnly)
+  case object AlignmentNotSame extends WarnID; forceConstruction(AlignmentNotSame)
+  case object InconsistentLengthKind extends WarnID; forceConstruction(InconsistentLengthKind)
+  case object FacetExplicitLengthOutOfRange extends WarnID; forceConstruction(FacetExplicitLengthOutOfRange)
+  case object DeprecatedFunctionDAFError extends WarnID; forceConstruction(DeprecatedFunctionDAFError)
+  case object DeprectatedBuiltInFormats extends WarnID; forceConstruction(DeprectatedBuiltInFormats)
+  case object DeprecatedEncodingNameUSASII7BitPacked extends WarnID; forceConstruction(DeprecatedEncodingNameUSASII7BitPacked)
+  case object UnsupportedAttributeSchemaLocation extends WarnID; forceConstruction(UnsupportedAttributeSchemaLocation)
+  case object UnsupportedAttributeBlockDefault extends WarnID; forceConstruction(UnsupportedAttributeBlockDefault)
+  case object UnsupportedAttributeFinalDefault extends WarnID; forceConstruction(UnsupportedAttributeFinalDefault)
+  case object ChoiceInsideHiddenGroup extends WarnID; forceConstruction(ChoiceInsideHiddenGroup)
+  case object DeprecatedPropertySeparatorPolicy extends WarnID; forceConstruction(DeprecatedPropertySeparatorPolicy)
+  case object PathNotToArray extends WarnID; forceConstruction(PathNotToArray)
+  case object RegexPatternZeroLength extends WarnID; forceConstruction(RegexPatternZeroLength)
```

Comment:
Might be worth sorting this list for readability (e.g. all the deprecated things would be together).

Reply:
And in dafext.xsd. Really this should be generated from dafext.xsd, but I'll leave that for some other developer dabbling in code generation. 

I will alphabetize this anyway for easier maintenance.